### PR TITLE
add filepath infix wildcard support

### PR DESCRIPTION
**Pull request by bowerscd**
Tested with this test set:

```
test case: pattern */some/path, filepath: /some/path expect: PASS
test case: pattern */some/path, filepath: /under/everything/some/path expect: PASS
test case: pattern /some/path*, filepath: /some/path expect: PASS
test case: pattern /some/path*, filepath: /some/path/everything/above expect: PASS
test case: pattern /exact/path, filepath: /exact/path expect: PASS
test case: pattern /exact/*/path, filepath: /exact/some/path expect: PASS
test case: pattern /exact/**/path, filepath: /exact/some/path expect: PASS
test case: pattern /exact/*/path*, filepath: /exact/some/path expect: PASS
test case: pattern *, filepath: /exact/some/path expect: PASS
test case: pattern */*/*, filepath: /u//other/doesn-t-matter/some/path.lib expect: PASS
test case: pattern /exact/*/exact/*/test, filepath: /exact//exact//test expect: PASS
test case: pattern /*, filepath: /* expect: PASS
test case: pattern /exact/*/exact/*/test, filepath: /exact/exact/test expect: FAIL
test case: pattern /, filepath: /* expect: FAIL
test case: pattern /exact/path, filepath: /unmmatched expect: FAIL
test case: pattern */soue/path, filepath: /some/path expect: FAIL
test case: pattern */everything/some, filepath: /under/everything/some/path expect: FAIL
test case: pattern /some/pathe*, filepath: /some/path expect: FAIL
test case: pattern /some/pav*, filepath: /some/pathe expect: FAIL
```

### CODE REVIEW

File: security/ipe/ipe-property.c [hunk 1]
```diff
@@ -4,17 +4,18 @@
 #include <linux/mount.h>
 #include <linux/dcache.h>
 #include <linux/binfmts.h>
+#include <linux/string.h>
 #include "ipe.h"
 #include "ipe-property.h"
 
-#define ARBITRARY_FILEPATH_MAX 256
+#define MAX_FILEPATH 256
```
Comment (jaskarankhurana) [inline security/ipe/ipe-property.c]: Can we check if we can create a file bigger than this.

File: security/ipe/ipe-property.c [hunk 2]
```diff
@@ -37,51 +38,64 @@ int ipe_get_filepath(struct ipe_operation_ctx *ctx, struct path *path)
...
+
+  // copy for strsep - stack buffer
+  // to avoid allocation failure
+  strcpy(localPattern, pattern);
+
+  subPattern = strsep(&temp, "*");
+  while(subPattern != NULL)
+  {
+    // locate substring
+    substring = strstr(&ctx->filepath[fpEnd], subPattern);
```
Comment (jaskarankhurana) [inline security/ipe/ipe-property.c]: Can you take a pointer in the start to ctx->filepath and use that

File: security/ipe/ipe-property.c [hunk 3]
```diff
@@ -37,51 +38,64 @@ int ipe_get_filepath(struct ipe_operation_ctx *ctx, struct path *path)
...
+    if(NULL == substring)
+    {
+      return false;
+    }
+
+    // keep track of where we are in
+    // the filepath
+    fpStart = substring - ctx->filepath;
+	fpEnd = fpStart + strlen(subPattern);
```
Comment (jaskarankhurana) [inline security/ipe/ipe-property.c]: tabs

File: security/ipe/ipe-property.c (hunk 2 above)
Comment (bowerscd) [reply security/ipe/ipe-property.c]: you mean you want

&ctx->filepath + fpEnd?

File: security/ipe/ipe-property.c [hunk 4]
```diff
@@ -37,51 +38,64 @@ int ipe_get_filepath(struct ipe_operation_ctx *ctx, struct path *path)
...
-		// Postfix Wildcard
-		--len;
-
-		// first len characters must match
-		return strncmp(ctx->filepath, property, len) == 0;
-	}
-
-	// Direct Compare - all characters must match
-	return strcmp(ctx->filepath, property) == 0;
+  char localPattern[MAX_FILEPATH] = { 0 };
```
Comment (jaskarankhurana) [inline security/ipe/ipe-property.c]: the tabs seem off for this whole function. It should be 8.

File: security/ipe/ipe-property.c [hunk 5]
```diff
@@ -37,51 +38,64 @@ int ipe_get_filepath(struct ipe_operation_ctx *ctx, struct path *path)
...
+
+    // keep track of where we are in
+    // the filepath
+    fpStart = substring - ctx->filepath;
+	fpEnd = fpStart + strlen(subPattern);
+
+    // keep a record of the previous match
+    // for checking the termination of the
+    // filepath
+    previousPattern = subPattern;
```
Comment (jaskarankhurana) [inline security/ipe/ipe-property.c]: Use sub_pattern style names.

File: security/ipe/ipe-property.c [hunk 6]
```diff
@@ -37,51 +38,64 @@ int ipe_get_filepath(struct ipe_operation_ctx *ctx, struct path *path)
...
+  char* temp;
+  const char *substring;
+  size_t fpStart = 0;
+  size_t fpEnd = 0;
+
+  // copy for strsep - stack buffer
+  // to avoid allocation failure
+  strcpy(localPattern, pattern);
+
+  subPattern = strsep(&temp, "*");
```
Comment (jaskarankhurana) [inline security/ipe/ipe-property.c]: IS temp even pointing to something?

**Pull request closed without merging**